lib/stm32/f1/flash.c

```diff
@@ -299,5 +299,27 @@ void flash_erase_all_pages(void)
 	FLASH_CR2 &= ~FLASH_CR_MER;
 }
 
+/** @brief Get size in bytes of one flash page
+*/
+uint32_t flash_get_page_size(void)
+{
+	uint32_t idcode = MMIO32(DBGMCU_BASE) & 0x7FF;
+	uint32_t page_size = 0;
+	
+	switch (idcode)
+	{
+		case 0x412:
+		case 0x410:
+			page_size = 1024;
+			break;
+		case 0x414:
+		case 0x418:
+		case 0x430:
```

Comment:
at the very least, you want all these cases to be numerically ordered.

Reply:
Also, you're missing the f100 line, 0x420 and 0x428 (at least)

Reply:
These are the only IDs I found in the manual. Where did you find the others?

Reply:
there's more than one manual :) if you're only looking at rm0008, you're not covering all of the f1s....  go back to st's website, click on "f1" and then click on docs, you can see the others :)